packages/test/src/test-integration-update-interceptors.ts

```diff
@@ -0,0 +1,79 @@
+import { WorkflowStartUpdateInput, WorkflowStartUpdateOutput } from '@temporalio/client';
+import * as wf from '@temporalio/workflow';
+import { Next, UpdateInput, WorkflowInboundCallsInterceptor, WorkflowInterceptors } from '@temporalio/workflow';
+import { helpers, makeTestFunction } from './helpers-integration';
+
+const update = wf.defineUpdate<string[], [string]>('update');
+
+export async function workflowWithUpdates(): Promise<string[]> {
+  const state: string[] = [];
+  const updateHandler = async (arg: string): Promise<string[]> => {
+    state.push(arg);
+    if (arg === 'fail-update') {
+      throw new wf.ApplicationFailure(`Deliberate ApplicationFailure in handler`);
+    }
+    return state;
+  };
+  wf.setHandler(update, updateHandler);
+  await wf.condition(() => state.includes('done'));
```

Comment:
Why make this workflow so elaborated if you only test startUpdate and executeUpdate?

Reply:
Are there some simplifications that you see that can be made? The `await wf.condition()` here is necessary to prevent the workflow exiting before the update is handled.

> if you only test startUpdate and executeUpdate?

The workflow allows us to implement 9 different tests:

```
'Update can be executed via executeUpdate()'
'Update can be executed via startUpdate() and handle.result()'
'Update validator can reject when using executeUpdate()'
'Update validator can reject when using handle.result() but handle can be obtained without error'
'Update: ApplicationFailure in handler rejects the update'
'Update is rejected if there is no handler'
'Update sent after workflow completed'
'Update id can be assigned and is present on returned handle'
```

so it has the ability to throw an `ApplicationFailure`, etc. I did investigate writing different workflows for different tests, but it was much more code, and a much higher mental burden to comprehend the full set of tests.